java-checks/src/main/java/org/sonar/java/checks/naming/BadMethodNameCheck.java

```diff
@@ -57,16 +57,22 @@ public void setContext(JavaFileScannerContext context) {
     super.setContext(context);
   }
 
-
   @Override
   public void visitNode(Tree tree) {
     MethodTree methodTree = (MethodTree) tree;
-    if (isNotOverriden(methodTree) && !pattern.matcher(methodTree.simpleName().name()).matches()) {
+    if (!isExcluded(methodTree) && !pattern.matcher(methodTree.simpleName().name()).matches()) {
       reportIssue(methodTree.simpleName(), "Rename this method name to match the regular expression '" + format + "'.");
     }
   }
 
-  private static boolean isNotOverriden(MethodTree methodTree) {
-    return Boolean.FALSE.equals(methodTree.isOverriding());
+  private boolean isExcluded(MethodTree methodTree) {
```

Comment:
This method can be `static`
```suggestion
  private static boolean isExcluded(MethodTree methodTree) {
```